Lib/test/test_unittest/test_result.py

```diff
@@ -275,6 +275,39 @@ def get_exc_info():
         self.assertEqual(len(dropped), 1)
         self.assertIn("raise self.failureException(msg)", dropped[0])
 
+    def test_addFailure_filter_traceback_frames_chained_exception_cycle(self):
+        class Foo(unittest.TestCase):
+            def test_1(self):
+                pass
+
+        def get_exc_info():
```

Comment:
I think we should also have a test for the edge case of a single exception in a loop with itself.

Reply:
Addressed in 6b868df